crates/key-server/src/server.rs

```diff
@@ -674,41 +650,67 @@ async fn main() -> Result<()> {
     } else {
         Hex::decode(&master_key).expect("MASTER_KEY should be hex encoded")
     };
-    // TODO: remove this when the legacy key server is no longer needed
-    let legacy_object_id =
-        env::var("LEGACY_KEY_SERVER_OBJECT_ID").expect("LEGACY_KEY_SERVER_OBJECT_ID must be set");
-    let object_id = env::var("KEY_SERVER_OBJECT_ID").expect("KEY_SERVER_OBJECT_ID must be set");
-    let network = env::var("NETWORK")
-        .map(|n| Network::from_str(&n))
-        .unwrap_or(Network::Testnet);
+
+    // If CONFIG_PATH is set, read the configuration from the file.
+    // Otherwise, use the legacy environment variables.
+    let options: KeyServerOptions = match env::var("CONFIG_PATH") {
+        Ok(config_path) => {
+            info!("Loading config file: {}", config_path);
+            serde_yaml::from_reader(
```

Comment:
this one would ignore unknown fields in the yaml file, right? i think that's ok, maybe let's document here that we do that

Reply:
No, it fails on unknown fields. Adding a test to make it clear.